traffic_portal/app/src/common/modules/dialog/input/dialog.input.tpl.html

```diff
@@ -24,7 +24,7 @@ <h4 class="modal-title">{{::params.title}}</h4>
 <div class="modal-body">
     <p ng-bind-html="params.message"></p>
     <form name="inputForm" novalidate>
-        <input type="text" class="form-control" ng-model="inputValue" ng-maxlength="256">
+        <input type="{{params.type ? params.type : 'text'}}" class="form-control" ng-model="inputValue" ng-maxlength="256">
```

Comment:
This will not enforce a min or max on numeric inputs, although your message states that log days should be between 1 and 365.

Also `ng-maxlength="{{some constant value}}"` can just be `maxlength="{{some constant value}}"`

Reply:
right but this is a generic component so didn't want to overload it.

Reply:
I was going to suggest you use an `*ngIf`-`else` to insert different inputs depending on the type, but it turns out that only works in Angular, not AngularJS. In AngularJS you'd just need to do two `ng-if`s which is kind of a cludgy solution. So never mind, I guess.